internal/plugin/plugin_manager_helper.go

```diff
@@ -86,51 +87,81 @@ func BuildManifestMap(log *logrus.Logger, name string, repoName string) (map[str
 }
 
 // takes url as input and returns index.yml for plugin repository
-func GetYamlFromUrl(url string) (map[string]interface{}, error) {
-	res, err := http.Get(url)
-	if err != nil {
-		return nil, err
-	}
+func GetYamlFromUrl(URL string) (map[string]interface{}, error) {
+	var manifest map[string]interface{}
+	if !IsUrl(URL) {
+		index, err := ioutil.ReadFile(URL)
+		if err != nil {
+			return nil, err
+		}
+		err = yaml.Unmarshal(index, &manifest)
+		if err != nil {
+			return nil, err
+		}
+		return manifest, nil
+	} else {
+		res, err := http.Get(URL)
+		if err != nil {
+			return nil, err
+		}
 
-	defer res.Body.Close()
+		defer res.Body.Close()
 
-	body, err := ioutil.ReadAll(res.Body)
+		body, err := ioutil.ReadAll(res.Body)
 
-	if err != nil {
-		return nil, err
-	}
-	var manifest map[string]interface{}
-	err = yaml.Unmarshal(body, &manifest)
-	if err != nil {
-		panic(err)
+		if err != nil {
+			return nil, err
+		}
+
+		err = yaml.Unmarshal(body, &manifest)
+		if err != nil {
+			return nil, err
+		}
 	}
 	return manifest, nil
 }
 
 // takes url as input and fetches the manifest of a plugin
-func YamlToManifest(url string) (Manifest, error) {
+func YamlToManifest(URL string) (Manifest, error) {
 	plugin := Manifest{}
-	res, err := http.Get(url)
-	if err != nil {
-		return plugin, err
-	}
+	if !IsUrl(URL) {
```

Comment:
I personally think that instead of having `IsUrl` in each of these functions for handling 1) the index, 2) the manifest, 3) the binary that you could instead create a new function `GetBytes(URL string) ([]byte, error)` for example that just takes the URL (whether that be `file:///some/path`, `https://raw.github...`, `/some/path`) and returns the bytes.